gatsby/create-node/create-preview-cards.js

```diff
@@ -0,0 +1,68 @@
+const slugify = require('slugify');
+const path = require('path');
+const { siteMetadata } = require('../../gatsby-config');
+const {
+  generateCard,
+} = require('../util/preview-card-generator/generate-card');
+
+/**
+ * Create a preview card file and put the url to the Gatsby store to be able
+ * to link it in the page header.
+ */
+const createPreviewMarkdown = ({ node, _, actions }) => {
+  const { createNodeField } = actions;
+  const post = node.frontmatter;
+
+  // we need the title in order to generate anything
+  if (!post.title) {
+    return;
+  }
+
+  // we create a file name from the path (which is the page slug and more unique) but if none is available take the title
+  const fileName = `social-card---${slugify(post.path || post.title, {
+    lower: true,
+  })}.jpg`;
+  const outputFile = path.join('public', fileName);
+  const publicUrl = `${siteMetadata.siteUrl}/${fileName}`;
+
+  generateCard({ title: post.title }, outputFile).then(() => {
+    createNodeField({
+      node,
+      name: `socialCard`,
+      value: publicUrl,
+    });
+  });
+};
+
+const createPreviewJob = ({ node, _, actions }) => {
+  const { createNodeField } = actions;
+
+  const fileName = `social-card---career-${slugify(node.name, {
+    lower: true,
+  })}.jpg`;
+
+  const outputFile = path.join('public', fileName);
+  const publicUrl = `${siteMetadata.siteUrl}/${fileName}`;
+
+  generateCard({ title: node.name }, outputFile).then(() => {
+    createNodeField({
+      node,
+      name: 'socialCard',
+      value: publicUrl,
+    });
+  });
+};
```

Comment:
the logic to create a preview card for a given job position. This is again not part of the source plugin. On the other hand seeing this and the path handling which are both attached on `onCreateNode`. Would this be a seprate plugiin (in an ideal world) ? Or would we marry the source plugin with those two responsibilities and remove the `source` part and call it our career plugin ? I like it more like that for now as it's like we would use any other source plugin.

Reply:
it wouldn't be a "source" plugin, but a normal plugin. 

we could create a "gatsby-plugin-preview-image" plugin and share it on the gatsby plugins page